athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/MetadataHandler.java

```diff
@@ -426,6 +431,7 @@ public abstract GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplit
      */
     public PingResponse doPing(PingRequest request)
     {
+        checkBucketAuthZ(AmazonS3ClientBuilder.standard().build(), spillBucket);
```

Comment:
Ping is meant to be light-weight so that it aids in both discovery and cold-start. I don't think we should apply this from PING. What are your thoughts on doing this from getSplits which is the 1st place that the spill bucket is potentially used?

Reply:
There's a tradeoff. If a customer's request will result in spilling, then checking as early as possible avoids wasting resources before spilling fails. However, an unauthorized bucket will prevent a customer from getting the response even when the response is in-memory

Reply:
I've done some experiments. Checking bucket ownership takes around 600 ms for an account that owns 25 buckets. This brings the total doPing time up to 620 ms. In terms of latency, checking bucket ownership is not "light-weight".

If this is what you are concerned about, I can move it to getSplits. What are your thoughts on potentially wasting some resources before checking bucket ownership?

Reply:
getSplits happens pretty early in the query lifecycle so I'm ok with it. We also don't expect this violation to happen very often. When combined with the caching I mentioned below it should be fine.

Reply:
I'll move it to getSplits and add caching. Thanks for the comment, Anthony!